src/test/java/com/atlassian/db/replica/api/AuroraMultiReplicaConsistencyTest.java

```diff
@@ -105,4 +119,33 @@ void shouldCloseReplicaConnectionsWhenNotConsistent() throws SQLException {
             verify(replica).close();
         }
     }
+
```

Comment:
you mentioned that lifecycle managment was impacted.
I wonder whether this shouldn't be tested here?

Reply:
`shouldNotOpenConnectionIfNotNeeded` tests it.